### ISSUE TITLE
0.7.50 hang on loading

### ISSUE THREAD

**Issue by Exovi** [ka-dyn/ExpandedContent#117]
**Describe the bug**
Recent update of Expanded Content causes WotR to hang on loading.
Reverting to 7.4 fixed it!

**Log**
[332.4271 - Unity]: NullReferenceException: Object reference not set to an instance of an object
ExpandedContent.Tweaks.Archetypes.DivineTracker.AddDivineTracker () (at <cd5fa2bc4ffa4e53b60971bfa7e5b4e9>:0)
ExpandedContent.Tweaks.ContentAdder+BlueprintsCache_Init_Patch.Postfix () (at <cd5fa2bc4ffa4e53b60971bfa7e5b4e9>:0)
(wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Init_Patch182(Kingmaker.Blueprints.JsonSystem.BlueprintsCache)
(wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.StartGameLoader.Kingmaker.Blueprints.JsonSystem.StartGameLoader.LoadPackTOC_Patch14(Kingmaker.Blueprints.JsonSystem.StartGameLoader)
Kingmaker.GameStarter+<StartGameCoroutine>d__25.MoveNext () (at <1bdbbfc0e6fc4f2db83179edfe399546>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext () (at <4fae2288737c4dfda349689bbefabfcc>:0)

**Additional context**
I have quite a few other mods. If there is an easy way to send a list, let me know and I will

**Comment by bittercranberry** [comment]
Please make sure all mods are up to date.

If that doesn't fix it then could you post the full log as an attachment please?

**Comment by fanofstuff** [comment]
Hi, I actually had the same issue - just returned to WotR, updated everything, game hung on load. 

Figured I'd do my due-diligence, looked in the logs, saw the same error, disabled Expanded Content while keeping all other mods, game loaded fine. Went back, disabled all other mods besides TTT Core + TTT Base + Expanded Content, same hang as before. 

Hopefully the attached log file will help - it's from my last test. Hope this helps - I'll be using the previous version in the meantime! 
[GameLogFull.txt](https://github.com/ka-dyn/ExpandedContent/files/12733089/GameLogFull.txt)

**Comment by bittercranberry** [comment]
I have just tried this mod combo and the game loads fine for me, which make troubleshooting on my end a bit of a pain.

Please use Steam to verify your game files, then redownload the 3 mods from the modpages or modfinder (in case the download corrupted).

After if the game still hangs open UMM in game (Default, CTRL + F10) then on the second tab at the bottom you should have a full log button. That log should show me where the load gets stuck.

**Comment by bittercranberry** [comment]
Most tickets like this end with the user fixing the issue and now telling anyone, if there are no responses after a week I'll close the ticket.

**Comment by Christoph-Wagner** [comment]
Same issue here, I have a lot of mods installed, but hope this is still useful:

[ummdetailedlog.txt](https://github.com/ka-dyn/ExpandedContent/files/12767982/ummdetailedlog.txt)


The relevant part is here:

```
[ExpandedContent] Added: f580c9180f644a228f633396b1536838 - DivineTrackerArchetype
[ExpandedContent] Added: 74bf574973a64d9b9413d2cd29346acd - DivineTrackerFavoredWeapon
Uploading Crash Report
NullReferenceException: Object reference not set to an instance of an object
  at ExpandedContent.Tweaks.Archetypes.DivineTracker.AddDivineTracker () [0x0008f] in <cd5fa2bc4ffa4e53b60971bfa7e5b4e9>:0 
  at ExpandedContent.Tweaks.ContentAdder+BlueprintsCache_Init_Patch.Postfix () [0x0010d] in <cd5fa2bc4ffa4e53b60971bfa7e5b4e9>:0 
  at (wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Init_Patch155(Kingmaker.Blueprints.JsonSystem.BlueprintsCache)
  at (wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.StartGameLoader.Kingmaker.Blueprints.JsonSystem.StartGameLoader.LoadPackTOC_Patch6(Kingmaker.Blueprints.JsonSystem.StartGameLoader)
  at Kingmaker.GameStarter+<StartGameCoroutine>d__25.MoveNext () [0x00375] in <1bdbbfc0e6fc4f2db83179edfe399546>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00026] in <4fae2288737c4dfda349689bbefabfcc>:0 
```

**Comment by bittercranberry** [comment]
Thanks I can work with that, I'll put some extra logging in the mod and release that as an update tonight. Could you please verify the game files on Steam and download my update if it is out, then post the new log if you crash?

**Comment by bittercranberry** [comment]
[ExpandedContent.zip](https://github.com/ka-dyn/ExpandedContent/files/12769819/ExpandedContent.zip)

If you install this it has better logging on where you crashed. Hopefully I can sort this for you.

Also I'm not sure why/how you are loading TTT before EC but I don't think that's causing this.

**Comment by Christoph-Wagner** [comment]
Sadly, it doesn’t seem to log any more than before. FWIW, I had a pretty outdated UMM version and am now updated to 0.27.8.0, but that didn’t change anything :/ All mods are updated as far as possible (before, only dark codex was not).

[Player.log](https://github.com/ka-dyn/ExpandedContent/files/12774098/Player.log)

**Comment by bittercranberry** [comment]
No logs is good for figureing it out, it means you fail loading Irori for some reason.
I'll look into it tonight but in the meantime go to this Discord  https://discord.com/channels/645948717400064030/815735034514112512
The people there are wizards at sorting stuff like this.

**Comment by Christoph-Wagner** [comment]
Glad it helped. And I’m good, I’m only playing through the ashes in anticipation of the next DLC and will be fine without your mod, just encountered the issue, found the existing one, and decided to give the logs asked for that the others didn’t ;)

**Comment by bittercranberry** [comment]
Turns out it is being caused by TTT-Base loading before EC, uninstalling both and then reinstalling should fix the issue.

**Comment by Christoph-Wagner** [comment]
Sorry for the late reply. Just tried uninstalling both and then reinstalling, and then uninstalling just TTT-Base and reinstalling it. No matter what, TTT-Base loads first. 

Probably not relevant, but here’s the error with everything updated:

```
[ExpandedContent] Added: 36d75d0c41fb497c98bf3e07a1fe8b2e - NaderiFeature
Uploading Crash Report
NullReferenceException: Object reference not set to an instance of an object
  at ExpandedContent.Tweaks.Deities.PatchPulura.AddPulura () [0x000dd] in <74fc05199bae4fe480eca21a0ad432da>:0 
  at ExpandedContent.Tweaks.ContentAdder+BlueprintsCache_Init_Patch.Postfix () [0x00491] in <74fc05199bae4fe480eca21a0ad432da>:0 
  at (wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Kingmaker.Blueprints.JsonSystem.BlueprintsCache.Init_Patch155(Kingmaker.Blueprints.JsonSystem.BlueprintsCache)
  at (wrapper dynamic-method) Kingmaker.Blueprints.JsonSystem.StartGameLoader.Kingmaker.Blueprints.JsonSystem.StartGameLoader.LoadPackTOC_Patch6(Kingmaker.Blueprints.JsonSystem.StartGameLoader)
  at Kingmaker.GameStarter+<StartGameCoroutine>d__25.MoveNext () [0x00375] in <1bdbbfc0e6fc4f2db83179edfe399546>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00026] in <4fae2288737c4dfda349689bbefabfcc>:0 
```

**Comment by bittercranberry** [comment]
I've done a hotfix, please redownload from github and tell me if it works.
If it does it will be a proper update later today, if not I'll give up for a week as I'm still convinced this is some weird setup error.

**Comment by bittercranberry** [comment]
[ExpandedContent.zip](https://github.com/ka-dyn/ExpandedContent/files/12840579/ExpandedContent.zip)
Does this fix it?

**Comment by Christoph-Wagner** [comment]
It does indeed fix it! It loads before TTT-Base now (besides the compat patches that load later), and the error is gone :) Thank you :)

**Comment by bittercranberry** [comment]
Fixed in 0.7.62
